src/rtctools/optimization/multi_seed_mixin.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Dict, Iterable
+
+from rtctools.optimization.optimization_problem import OptimizationProblem
+from rtctools.optimization.timeseries import Timeseries
+
+
+class MultiSeedMixin(OptimizationProblem):
+    """
+    Enables a workflow to solve an optimization problem by trying multiple seeds.
+    """
+
+    def __init__(self, **kwargs):
+        self.__selected_seed_id = None
+        super().__init__(**kwargs)
+
+    def use_seed_id(self) -> bool:
+        """Return ``True`` if the selected seed id is used.
+
+        Return ``True`` if the seed corresponding to the selected seed identifier is used
+        for the current run.
+        By default, this is ``True``.
+        Overwrite this method when in some cases the seed is not based on the seed id,
+        e.g. in case of goal programming, return False after the first priority,
+        or in case of homotopy, return ``False`` after the homotopy parameter is increased.
+        """
+        return True
+
+    def selected_seed_id(self) -> Any:
```

Comment:
Do we need this public method? Isn't it sufficient to maintain the private `__selected_seed_id` variable?

Reply:
Perhaps not. I'll see if this method can be avoided.

Reply:
In some cases, we might want to use different solver parameters depending on the seed. For example, when using homotopy, we might want to start immediately with the nonlinear case (theta_start = 1), while for the default seed, we might want to start with the linear case (theta_start = 0). Exposing the selected seed id allows the user to adapt their solver options based on the seed. This behaviour is also checked in the SeedMixin test.

Reply:
why not override solver_options() and return the desired solver depending on the current value of theta? If it is already possible to implement the desired behavior in a lean way, we should not implement a new way to do the same.

Reply:
The case we had in mind, and which is included the test, has theta_start depend on the seed, not the other way around. We solve this by overwriting the homotopy options and let theta_start depend on the seed_id.